### ISSUE TITLE
add documentation on for new LearningRateScheduler interface

### ISSUE THREAD

**Issue by DSLituiev** [allenai/allennlp#3579]
I am refactoring old code and see that `LearningRateWithoutMetricsWrapper` has been removed, and no documentation on how to use new interface available.

I have a following code:
```
from torch.optim.lr_scheduler import MultiStepLR
from allennlp.training.learning_rate_schedulers import LearningRateWithoutMetricsWrapper

learning_rate_scheduler = LearningRateWithoutMetricsWrapper(MultiStepLR(optimizer,
                                                                        [10, 20, 40],
                                                                        gamma=0.25, last_epoch=-1))
```

 Looking into the [commits](https://github.com/allenai/allennlp/pull/2345/commits/12085f62ee0253dca224b77f9640d1f411ed1273), I guessed it must be something like:

```
from torch.optim.lr_scheduler import MultiStepLR
from allennlp.training.learning_rate_schedulers import LearningRateScheduler

optimizer = optim.SGD(model.parameters(), lr=0.1)
learning_rate_scheduler = LearningRateScheduler(optimizer, dict(type="MultiStepLR",
                                                                milestones=[10, 20, 40],
                                                                # initial_lr=0.1,
                                                                gamma=0.25, last_epoch=-1))
```

But that fails with error (whether `initial_lr=0.1` is included or not):
```
/Applications/anaconda3/envs/nlp/lib/python3.7/site-packages/allennlp/training/learning_rate_schedulers/learning_rate_scheduler.py in __init__(self, optimizer, last_epoch)
     12 class LearningRateScheduler(Scheduler, Registrable):
     13     def __init__(self, optimizer: torch.optim.Optimizer, last_epoch: int = -1) -> None:
---> 14         super().__init__(optimizer, "lr", last_epoch)
     15 
     16     def get_values(self) -> None:

/Applications/anaconda3/envs/nlp/lib/python3.7/site-packages/allennlp/training/scheduler.py in __init__(self, optimizer, param_group_field, last_epoch)
     42                 if self._initial_param_group_field not in group:
     43                     raise KeyError(
---> 44                         f"{self._initial_param_group_field} missing from param_groups[{i}]"
     45                     )
     46         self.base_values = [

KeyError: 'initial_lr missing from param_groups[0]'
```

**Comment by dirkgr** [comment]
Try creating the learning rate scheduler like this:
```
LearningRateScheduler.from_params(
    optimizer, dict(
        type="MultiStepLR",
        milestones=[10, 20, 40],
        initial_lr=0.1,
        gamma=0.25, last_epoch=-1))
```

**Comment by DSLituiev** [comment]
thank you @dirkgr , it did not work:

```
from allennlp.training.learning_rate_schedulers import LearningRateScheduler

optimizer = optim.SGD(model.parameters(), lr=0.1)
learning_rate_scheduler = LearningRateScheduler.from_params(
    optimizer, dict(
        type="MultiStepLR",
        milestones=[10, 20, 40],
        initial_lr=0.1,
        gamma=0.25, last_epoch=-1))
```

```

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-22-bd878d588bcb> in <module>
     13         milestones=[10, 20, 40],
     14         initial_lr=0.1,
---> 15         gamma=0.25, last_epoch=-1))
     16 
     17 

/Applications/anaconda3/envs/nlp/lib/python3.7/site-packages/allennlp/training/learning_rate_schedulers/learning_rate_scheduler.py in from_params(cls, optimizer, params)
     21     def from_params(cls, optimizer: torch.optim.Optimizer, params: Params):  # type: ignore
     22 
---> 23         scheduler_type = params.pop_choice("type", LearningRateScheduler.list_available())
     24         scheduler = LearningRateScheduler.by_name(scheduler_type)(
     25             optimizer, **params.as_dict()

AttributeError: 'dict' object has no attribute 'pop_choice'
```

```
import allennlp
allennlp.__version__
'0.9.1-unreleased'
```

**Comment by DSLituiev** [comment]
For now using this:

```
from torch.optim.lr_scheduler import MultiStepLR

from allennlp.training.learning_rate_schedulers.learning_rate_scheduler import _PyTorchLearningRateSchedulerWrapper


optimizer = optim.SGD(model.parameters(), lr=0.1)

learning_rate_scheduler = _PyTorchLearningRateSchedulerWrapper(MultiStepLR(optimizer,
                                                                        [10, 20, 40],
                                                                        gamma=0.25, last_epoch=-1))
```

I wish `_PyTorchLearningRateSchedulerWrapper` was a public method. Its usage is quite straightforward to me.

**Comment by dirkgr** [comment]
You're right, I think this is closer:
```
learning_rate_scheduler = LearningRateScheduler.from_params(
    optimizer, Params(dict(
        type="multi_step",
        milestones=[10, 20, 40],
        initial_lr=0.1,
        gamma=0.25, last_epoch=-1)))
```

**Comment by DSLituiev** [comment]
Where do I get `Params` from? Can you please provide imports?

Anyone here is able to explain how is this superior to simply making `_PyTorchLearningRateSchedulerWrapper` public? It seems much more complicated and requiring more syntactic acrobatics.

**Comment by dirkgr** [comment]
`Params` is `allennlp.common.Params`.

Maybe that is the right thing to do, but we would want that to be part of a better strategy for the whole library, making it easy to use programmatically.

**Comment by shimengfeng** [comment]
Hi there, 

I am still not sure how could I exactly using the scheduler. I am using 1.1 and it seems the from_params have been removed. Currently

```
model_parameters = [
        [n, p]
        for n, p in model.named_parameters() if p.requires_grad
    ]
optimizer = AdamOptimizer(model_parameters, lr=0.001)
```



Then I am not sure how could I use the `LearningRateScheduler ` or the code you provided above 

```
dict(
        type="multi_step",
        milestones=[10, 20, 40],
        initial_lr=0.1,
        gamma=0.25, last_epoch=-1)

```

I think I will need to add a param_groups in my optimizer, but I am not sure how exactly I can use it. I have visited this page: [https://docs.allennlp.org/master/api/training/optimizers/] and still confused. Thank you!
